src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1361,26 +1368,38 @@ private async Task ValidateDatabaseConnection()
                 Connection = conn
             };
 
-            string tablePrefix = GetTablePrefix(conn.Database, schemaName);
-            string queryPrefix = string.IsNullOrEmpty(tablePrefix) ? string.Empty : $"{tablePrefix}.";
+            string tableNameWithPrefix = GetTableNameWithPrefix(conn.Database, schemaName, tableName);
```

Comment:
For a connection string that has a databasename, conn.Database will not be empty, so tableNameWithPrefix will be for books e.g. `[testdb].[dbo].[books]` for mssql. The EntitiesDataSet dictionary will store the DataTable with this as the key. Subsequently, when you are trying to retrieve in the function above, you are looking up by `[dbo].[books]` How will the lookup succeed?

Reply:
I have totally removed the need for database name now. This was succeeding in our tests before due to lack of a database name being explicitly used in the test pipeline.